src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,30 @@
-import React from 'react';
-import { Loader } from '../Loader';
+import React, { useEffect, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { Loader } from '../Loader/Loader';
+import { getUser } from '../../api';
+
+interface Props {
+  todo: Todo;
+  closeModal: () => void;
+}
+
+export const TodoModal: React.FC<Props> = ({ todo, closeModal }) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoadingUser, setIsLoadingUser] = useState(true);
```

Comment:
It seems that you are initializing the state with props (todo.userId in this case). This is generally considered an anti-pattern in React. This is because the constructor or `useState` hook runs only once when the component is created, and then never runs again. This means that even if the prop updates, the state will not update. You should avoid copying props into state and instead use the prop directly.